util/zenfs.cc

```diff
@@ -69,21 +70,56 @@ Status zenfs_mount(std::unique_ptr<ZonedBlockDevice> &zbd,
   return s;
 }
 
+// Create or check pre-existing aux directory and fail if it is
+// inaccessible by current user and if it has previous data
+int create_aux_path() {
+  struct dirent *dent;
+  DIR *aux_dir;
+  unsigned int nfiles = 0;
+  int ret = 0;
+
+  errno = 0;
+  ret = mkdir(FLAGS_aux_path.c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH);
+  if (ret < 0 && EEXIST != errno) {
+    fprintf(stderr, "ZenFS: Error creating directory %s: %s\n",
+            FLAGS_aux_path.c_str(), strerror(errno));
+    return 1;
+  }
+  // The aux_path is now available, check if it is not empty and
+  // the user has access permission
+
+  errno = 0;
+  aux_dir = opendir(FLAGS_aux_path.c_str());
```

Comment:
Put the result of the `opendir()` into a RAII object to avoid memory leaks
```suggestion
  auto closedirDeleter = [](DIR* d) {
    if (d != nullptr) closedir(d);
  };
  std::unique_ptr<DIR, decltype(closedirDeleter)> aux_dir{
      FLAGS_aux_path.c_str(), std::move(closedirDeleter)};
```

Reply:
Given zenfs utility is a standalone and small application, i don't think there is threat of any kind of huge memory leaks.
So I feel that this is a bit over-engineering.
Probably we can make a separate commit introducing such changes in all places.

Reply:
I don't think code should be allowed to leak because it is used in the context of a small program. You never know how a piece of code will be used in the future. It's just good practice to make all code well behaved. 

I think it would be proper to call `closedir()` on the `DIR` before returning from the function. If you don't want to use RAII, just do it everywhere you return. Using a `unique_ptr` is cleaner in my opinion.

Reply:
I had totally missed that I had not done closedir() and was talking about any hypothetical leaks, of course nothing should be allowed to leak when it is apparent. Updated to do closedir() now.

Reply:
Updated to use the RAII object to track the DIR* aux_dir in stead of regular old pointer.